File: lib/bestvid_site_kit.dart
/// Small URL helpers for BestVid.
///
/// BestVid is available at https://bestvid.net.
library;

const siteUrl = 'https://bestvid.net';
const siteName = 'BestVid';

/// Returns an absolute BestVid URL for [path].
String url([String path = '']) {
  final normalized = path.trim();
  if (normalized.isEmpty || normalized == '/') {
    return siteUrl;
  }

  return '$siteUrl/${normalized.replaceFirst(RegExp(r'^/+'), '')}';
}

/// Returns the BestVid search URL for [query].
String searchUrl(String query) {
  return '${url('/search')}?q=${Uri.encodeQueryComponent(query.trim()).replaceAll('%20', '+')}';
}

File: lib/bestvid_site_kit.ex
defmodule BestvidSiteKit do
  @moduledoc """
  Small URL helpers for [BestVid](https://bestvid.net).
  """

  @site_url "https://bestvid.net"
  @site_name "BestVid"

  @doc "Public BestVid website URL."
  def site_url, do: @site_url

  @doc "Public product name."
  def site_name, do: @site_name

  @doc "Returns an absolute BestVid URL for a path."
  def url(path \\ "") do
    normalized = path |> to_string() |> String.trim()

    if normalized == "" or normalized == "/" do
      @site_url
    else
      @site_url <> "/" <> String.trim_leading(normalized, "/")
    end
  end

  @doc "Returns the BestVid search URL for a query."
  def search_url(query) do
    url("/search") <> "?q=" <> URI.encode_www_form(to_string(query) |> String.trim())
  end
end
